DepositApi/Controllers/CalculateDepositApiController.cs

```diff
@@ -13,35 +13,37 @@ namespace DepositApi.Controllers
 {
     [ApiController]
     [Route("/api/calculate")]
-    public class DepositCalcApiController : ControllerBase
+    public class DepositCalculationApiController : ControllerBase
     {
         private readonly IDepositService depositService;
         private readonly IMapper mapper;
 
-        public DepositCalcApiController(IDepositService depositService, IMapper mapper)
+        public DepositCalculationApiController(IDepositService depositService, IMapper mapper)
         {
             this.depositService = depositService;
             this.mapper = mapper;
         }
 
         [HttpGet]
-        public ActionResult Get(DepositModel deposit)
+        public async Task<ActionResult> Get(DepositModel deposit)
         {
             var validator = new DepositValidator();
             var valResult = validator.Validate(deposit);
             if (valResult.IsValid)
             {
                 var depositDTO = this.mapper.Map<DepositDTO>(deposit);
-                var depositCalcDTO = this.depositService.PersentCalculationAsync(depositDTO);
-                var depositCalc = this.mapper.Map<List<DepositCalcModel>>(depositCalcDTO);
-                return new JsonResult(depositCalc);
+                var depositCalculetionDTO = await this.depositService.PersentCalculationAsync(depositDTO);
+                var depositCalculetion = this.mapper.Map<List<DepositCalculationModel>>(depositCalculetionDTO);
+
+                return new JsonResult(depositCalculetion);
```

Comment:
You don't have to create JsonResult here, just return Ok(depositCalculation) here, and let the request pipeline to handle serialization.